lib/spack/spack/test/directory_layout.py

```diff
@@ -196,7 +196,10 @@ def test_handle_unknown_package(
         layout.create_install_directory(spec)
         installed_specs[spec] = layout.path_for_spec(spec)
 
-    spack.repo.swap(mock_db)
+    # Plug-in the mock repository
+    cache_real_repo = spack.repo[:]
```

Comment:
You shouldn't count only methods I implemented directly. :smile:  

I use `clear` (whose implementation is ported from python 3.6) and `append` instead of an in-place swap for instance. I use for loops over the repo. The point is: once you blend into the usual list syntax, you'll start using it.